### PR TITLE
Collada animation (visibility / tranlation / rotation / scale) implemented

### PR DESCRIPTION
**Pull request by SSPkrolik**

### CODE REVIEW

**Comment by yglukhov** [comment]
Does it work with JS?

**Comment by SSPkrolik** [comment]
@yglukhov There's a small problem in nimasset with JS

File: rod/node.nim [hunk 1]
```diff
@@ -200,11 +200,11 @@ proc `worldPos=`(n: Node, p: Vector3) =
 
 proc visitProperties*(n: Node, p: var PropertyVisitor) =
     p.visitProperty("name", n.name)
-    p.visitProperty("translation", n.translation)
+    p.visitProperty("translation", n.translation, { pfAnimatable })
```
Comment (yglukhov) [inline rod/node.nim]: this file should be completely reverted.

File: rod/scene_composition.nim [hunk 2]
```diff
@@ -250,7 +252,7 @@ proc setupFromColladaNode(cn: ColladaNode, colladaScene: ColladaScene): Node =
     for it in cn.children:
         result.addChild(setupFromColladaNode(it, colladaScene))
 
-proc loadSceneAsync*(resourceName: string, handler: proc(n: Node3D)) =
+proc loadSceneAsync*(resourceName: string, handler: proc(n: Node3D, a: seq[Animation] = @[])) =
```
Comment (yglukhov) [inline rod/scene_composition.nim]: Animations should already be registered in the node by name. The handler should know nothing about animations.

File: rod/scene_composition.nim [hunk 3]
```diff
@@ -262,4 +264,8 @@ proc loadSceneAsync*(resourceName: string, handler: proc(n: Node3D)) =
 
         let res = setupFromColladaNode(colladaScene.rootNode, colladaScene)
 
-        handler(res)
+        var animations: seq[Animation] = @[]
+        for anim in colladaScene.animations:
+            animations.add(animationWithCollada(res, anim))
```
Comment (yglukhov) [inline rod/scene_composition.nim]: Animations should be registered in the node, as it is done in ae_animation.

File: rod/edit_view.nim [hunk 4]
```diff
@@ -123,12 +124,11 @@ proc newTreeView(e: Editor, inspector: InspectorView): PanelView =
 
                 outlineView.expandRow(sip)
                 let path = callDialogFileOpen("Select file")
-                loadSceneAsync path, proc(n: Node) =
+                loadSceneAsync path, proc(n: Node, a: seq[Animation] = @[]) =
```
Comment (yglukhov) [inline rod/edit_view.nim]: This file should be completely reverted

File: editor/rodedit.nim [hunk 5]
```diff
@@ -42,24 +42,28 @@ proc startApplication() =
...
-    cameraNode.translation.z = 1
+    cameraNode.translation.z = 80
 
     let light = editView.rootNode.newChild("point_light")
     light.translation = newVector3(0,0,100)
     let lightSource = light.component(LightSource)
     lightSource.setDefaultLightSource()
 
-    loadSceneAsync "collada/balloons_test.dae", proc(n: Node) =
+    loadSceneAsync "collada/balloons_test.dae", proc(n: Node, a: seq[Animation] = @[]) =
```
Comment (yglukhov) [inline editor/rodedit.nim]: this line should be reverted.

File: editor/rodedit.nim [hunk 6]
```diff
@@ -42,24 +42,28 @@ proc startApplication() =
...
         editView.rootNode.addChild(n)
 
         mainWindow.addSubview(editView)
 
+        for anim in a:
```
Comment (yglukhov) [inline editor/rodedit.nim]: Node animation should be aquired with `animationNamed` declared in node.nim

File: rod/scene_composition.nim (hunk 3 above)
Comment (SSPkrolik) [inline rod/scene_composition.nim]: +

File: rod/scene_composition.nim (hunk 2 above)
Comment (SSPkrolik) [inline rod/scene_composition.nim]: +

File: rod/node.nim (hunk 1 above)
Comment (SSPkrolik) [inline rod/node.nim]: +

**Pull request merged**